docs/guides/transfer_learning.ipynb

```diff
@@ -0,0 +1,308 @@
+    "\n",
+    "Finally we have to define the optimiation procedure, for Transfer Learning we would like to train different parts of the model separately, there are a couple of options for this:\n",
+    "\n",
+    "1. Using `stop_gradient`.\n",
+    "2. Filtering the parameters for `jax.grad`.\n",
+    "3. Using multiple optimizers for different parameters.\n",
+    "\n",
+    "While each could be useful in different situations, we will recommend using multiple optimizers via [optax.multi_transform](https://optax.readthedocs.io/en/latest/api.html#optax.multi_transform) because its efficient and apart from freezing layers it also lets you implement differential learning rates by just changing the optimizers. To use `multi_transform` you have to do two things:\n",
+    "\n",
+    "1. Create a mappging between parameter partitions and their optimizer.\n",
+    "2. Create a pytree with the same shape as the parameters but its leaves containing the partition label.\n",
+    "\n",
+    "### Freezing layers\n",
+    "\n",
+    "To freeze layers with `multi_transform` we will create the `trainable` and `frozen` parameter partitions. For the `trainable` parameters we will use the `adam` optimizer, and for the `frozen` parameters we will use `optax.set_to_zero` which zeros-out the gradients. To map parameters to partitions we'll use the `traverse_util.path_aware_map` function, by leveraging the `path` argument we can map the `backbone` parameters to `frozen` and the rest to `trainable`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "FrozenDict({\n",
+       "    backbone: {\n",
+       "        embeddings: {\n",
+       "            class_embedding: 'frozen',\n",
+       "            patch_embedding: {\n",
+       "                kernel: 'frozen',\n",
+       "            },\n",
+       "        },\n",
+       "    },\n",
+       "    head: {\n",
+       "        bias: 'trainable',\n",
+       "        kernel: 'trainable',\n",
+       "    },\n",
+       "})"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from flax import traverse_util\n",
+    "import optax\n",
+    "\n",
+    "partition_optimizers = {'trainable': optax.adam(5e-3), 'frozen': optax.set_to_zero()}\n",
+    "param_partitions = freeze(traverse_util.path_aware_map(\n",
+    "  lambda path, v: 'frozen' if 'backbone' in path else 'trainable', variables['params']))\n",
+    "tx = optax.multi_transform(partition_optimizers, param_partitions)\n",
+    "\n",
+    "# visualize a subset of the param_partitions structure\n",
+    "flat = list(traverse_util.flatten_dict(param_partitions).items())\n",
+    "freeze(traverse_util.unflatten_dict(dict(flat[:2] + flat[-2:])))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Differential Learning Rates\n",
```

Comment:
Perhaps we can point to optax for "advanced" optimizer setups? This is a bit of a tangent wrt to transfer learning

Reply:
Ok, removed the example and replaced it with a short description of how to implement differential learning rates and a pointer to optax docs on advanced optimizers.